internal/flags/apiflags/flags.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiflags
+
+import (
+	"fmt"
+	"strings"
+
+	gencrd "github.com/operator-framework/operator-sdk/internal/generate/crd"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+	"github.com/spf13/pflag"
+)
+
+// AddAPIFlags ...
+type AddAPIFlags struct {
+	GitInit          bool
+	MakeVendor       bool
+	SkipValidation   bool
+	SkipGeneration   bool
+	GeneratePlaybook bool
+	APIVersion       string
+	Kind             string
+	OperatorType     string
+	Repo             string
+	HeaderFile       string
+	CrdVersion       string
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
+}
+
+// AddTo - Add the reconcile period and watches file flags to the the flagset
+// helpTextPrefix will allow you add a prefix to default help text. Joined by a space.
+func (f *AddAPIFlags) AddTo(flagSet *pflag.FlagSet) {
```

Comment:
Many of these flags are only relevant to `operator-sdk new` and should not be added to `operator-sdk add`. We should remove the following flags from the shared flagset:
- type
- repo
- git-init
- header-file
- vendor
- skip-validation

Reply:
You are saying that, we should not move all flags under `apiflags/flags.go`. rather, we should only have shared flagset under here, and still parse the args separately for the flags specific to 'new` under `new\cmd.go`. is this understanding correct?

Reply:
Yep :+1: